### ISSUE TITLE
the part of predict code inference time cost

### ISSUE THREAD

**Issue by muzi2045** [traveller59/second.pytorch#119]
```
if a_mask is not None:
    box_preds = box_preds[a_mask]
    cls_preds = cls_preds[a_mask]
if self._use_direction_classifier:
    if a_mask is not None:
        dir_preds = dir_preds[a_mask]
        # print(dir_preds.shape)
        dir_labels = torch.max(dir_preds, dim=-1)[1]
if self._encode_background_as_zeros:
    # this don't support softmax
    assert self._use_sigmoid_score is True
    total_scores = torch.sigmoid(cls_preds)
else:
    # encode background as first element in one-hot vector
    if self._use_sigmoid_score:
        total_scores = torch.sigmoid(cls_preds)[..., 1:]
    else:
        total_scores = F.softmax(cls_preds, dim=-1)[..., 1:]
```
this part code in the VoxelNet.py file looks like cost too much time in the network forward time ( I test this part code will cost 40ms ~ 50ms), there is any suggestion to speed up this part? thanks a lot

**Comment by muzi2045** [comment]
the most cost time part is
`box_preds = box_preds[a_mask]`
`cls_preds = cls_preds[a_mask]`
it look like the pytorch Tensor operator [ ] cost too much time. @traveller59

**Comment by traveller59** [comment]
This operation takes 50ms??? I know the anchor mask generation is slow so I disable it in newest code. but 50ms???
You need to call ```torch.cuda.synchronize()``` before and after your device code when you measure device code speed. for example:
```Python
t = time.time()
torch.cuda.synchronize()
# device code
torch.cuda.synchronize()
print(time.time() - t)
```

**Comment by muzi2045** [comment]
thanks a lot, that't my problem in measure the inference time
```
prepare cloud time:  0.0034172534942626953
  points reshape time: 1.0251998901367188e-05
  Start inference_vxnet...
  input preparation time: 0.006982088088989258
    example_convert_to_torch_time:  0.0033698081970214844
         voxel_feature_extractor_time: 0.007473468780517578
           middle_feature_extractor: 0.0024492740631103516
            rpn_time: 0.04791617393493652
        forward_time_cost:  0.06101799011230469
          predict_prepare_time: 0.0007412433624267578
            mask_time:  0.0008814334869384766
            torch.sigmoid_time: 4.601478576660156e-05
           predict_1:  0.0031690597534179688
           nms cost time: 0.004007101058959961
           predict_3_time:  0.0014758110046386719
          predict for once:  0.01109457015991211
        predict_time:  0.017305612564086914
      voxelnet inference time cost: 0.0799415111541748
    predict_time:  0.08209395408630371
  detection time: 0.0867319107055664
  remove low score box 0.00030159950256347656
  process predict box time:  0.00016307830810546875
total_detection_time:  0.09881854057312012
total callback time:  0.10401225090026855
```
that maybe the correct time cost , seem like the rpn layer cost the most time( the rpn layer is from second-v1.0)
